feat(auth): add retries to aws boto calls

samdammers opened the pull request:
* implement default retries for boto3
* implement default retries for stax api calls
* allow config of both of these configurations via Stax Config

`TOKEN_EXPIRY_THRESHOLD_IN_MINS` is an outlier in terms of configuring the Stax SDK. marked as deprecated in the README and will be removed in a future release. preference to configure via the Stax Config object (to which the consumer can pass in values as ENV vars should they wish to).

samdammers committed:
feat(auth): add retries to aws boto calls

the standard boto client defaults to 'legacy' which retries certain HTTP
status codes and a limited set of service errors. update retry config to
'standard' which has a broader set of errors/exceptions. in particular
it will retry 'TooManyRequestsException' which can be throttled in high
burst situations.

max retries can be configured by setting the 'STAX_API_AUTH_MAX_RETRIES'
environment var.

ZXYmania commented on README.md:
```diff
@@ -31,11 +31,12 @@ This can be seen in our [guide](https://github.com/stax-labs/lib-stax-python-sdk
+Allows configuration of the maximum number of attempts and the threshold to when the Auth library should re-cache the credentials
 *Suggested use when running within CI/CD tools to reduce overall auth calls*
 ~~~bash
+export STAX_API_AUTH_MAX_RETRIES=5
```
Can we make this a config setting?

samdammers replied:
Yes, refactored

samdammers committed:
feat(api): add default retry configuration to stax api calls

add default retries for stax api calls. allow these configuration to be
modified by consumers. amended prior boto3 configuration to opt for
storing in the Stax config object. marked existing token threshold ENV
configuration as deprecated.

samdammers committed:
remove debug code

samdammers committed:
chore(sdk): update comment with deprecated notice
